### ISSUE TITLE
Float weights in Kaplan-Meier

### ISSUE THREAD

**Issue by pzivich** [CamDavidsonPilon/lifelines#418]
There is some unexpected behavior when fitting a weighted KM with weights that are floats. Here is a quick toy example

`import pandas as pd`
`import lifelines`
`df = pd.DataFrame()`
`df['t'] = [1,2,3,4,5]`
`df['d'] = [1,1,1,1,1]`
`df['w1'] = [2,1,1,1,1]`
`df['w2'] = [2.1,0.9,0.9,0.9,0.9]`
When fitting integer weights (w1), everything behaves as expected. However, when floats are included instead (w2) some unexpected behavior occurs. Running the following lines:
`y = lifelines.KaplanMeierFitter().fit(df['t'],weights=df['w2'])`
`print(y.survival_function_)`
Produces 
`          KM_estimate`
`timeline             `
`0.0             1.000`
`1.0             0.375`
`2.0             0.375`
`3.0             0.375`
`4.0             0.375`
`5.0             1.000`

But what is expected is:
`          KM_estimate`
`timeline             `
`0.0             1.000`
`1.0             0.632`
`2.0             0.474`
`3.0             0.316`
`4.0             0.158`
`5.0             0.000`

I haven't figured out exactly where everything goes wrong, but it looks like it is due to y.event_table converting all of the counts to int32 rather than float32. Most likely something in _preprocess_inputs is rounding everything to integers for the event_table before the KM calculation. This would explain why the survival time goes to 1 at the last time point that is less than 1.

**Comment by CamDavidsonPilon** [comment]
Sorry I missed this - looks pretty important.

**Comment by pzivich** [comment]
No worries. I meant to look deeper into it and find where it goes wrong, but have been pretty bogged down lately

**Comment by pzivich** [comment]
Alright, I think I found where it goes wrong.  In lifelines/utils/__init__ the function survival_table_from_events() returns a pandas dataframe with all items converted to integers. It converts all the items to integers when the object is returned. Removing `.astype(int)` generates event_tables that have the correct integers when I tested the function locally. I don't know if removing this would have any downstream issues. I would guess integers was a stylistic choice?

Options are; (1) remove `.astype(int)` from a single line. Disadvantage is that event_tables won't look as clean and crisp (everything will have trailing zeroes). (2) add logic so that if weights are all integers then it is returned as integers. Otherwise, it leaves everything as a float
